server/src/domain/media/media.service.ts

```diff
@@ -59,51 +58,65 @@ export class MediaService {
         break;
       case AssetType.VIDEO:
-        const { videoStreams } = await this.mediaRepository.probe(asset.originalPath);
-        const mainVideoStream = this.getMainVideoStream(videoStreams);
-        if (!mainVideoStream) {
-          this.logger.error(`Could not extract thumbnail for asset ${asset.id}: no video streams found`);
-          return false;
-        }
-        const { ffmpeg } = await this.configCore.getConfig();
-        const config = { ...ffmpeg, targetResolution: thumbnail.jpegSize.toString(), twoPass: false };
-        const options = new ThumbnailConfig(config).getOptions(mainVideoStream);
-        await this.mediaRepository.transcode(asset.originalPath, jpegThumbnailPath, options);
-        this.logger.log(`Successfully generated video thumbnail ${asset.id}`);
+        await this.generateVideoThumbnail(asset, 'jpeg');
         break;
+      default:
+        throw new UnsupportedMediaTypeException(`Unsupported asset type for thumbnail generation: ${asset.type}`);
     }
 
-    await this.assetRepository.save({ id: asset.id, resizePath: jpegThumbnailPath });
-
+    const resizePath = this.ensureThumbnailPath(asset, 'jpeg');
+    await this.assetRepository.save({ id: asset.id, resizePath });
     return true;
   }
 
+  async generateImageThumbnail(asset: AssetEntity, format: 'jpeg' | 'webp') {
+    const { thumbnail } = await this.configCore.getConfig();
+    const size = format === 'jpeg' ? thumbnail.jpegSize : thumbnail.webpSize;
+    const thumbnailOptions = { format, size, colorspace: thumbnail.colorspace, quality: thumbnail.quality };
+    const path = this.ensureThumbnailPath(asset, format);
+    await this.mediaRepository.resize(asset.originalPath, path, thumbnailOptions);
+    this.logger.log(`Successfully generated ${format.toUpperCase()} thumbnail for asset ${asset.id}`);
+  }
+
+  async generateVideoThumbnail(asset: AssetEntity, format: 'jpeg' | 'webp') {
+    const { ffmpeg, thumbnail } = await this.configCore.getConfig();
+    const size = format === 'jpeg' ? thumbnail.jpegSize : thumbnail.webpSize;
+    const { videoStreams } = await this.mediaRepository.probe(asset.originalPath);
+    const mainVideoStream = this.getMainVideoStream(videoStreams);
+    if (!mainVideoStream) {
+      this.logger.warn(`Skipped thumbnail generation for asset ${asset.id}: no video streams found`);
+      return;
+    }
+    const path = this.ensureThumbnailPath(asset, format);
+    const config = { ...ffmpeg, targetResolution: size.toString() };
+    const options = new ThumbnailConfig(config).getOptions(mainVideoStream);
+    await this.mediaRepository.transcode(asset.originalPath, path, options);
+  }
+
   async handleGenerateWebpThumbnail({ id }: IEntityJob) {
     const [asset] = await this.assetRepository.getByIds([id]);
     if (!asset || !asset.resizePath) {
       return false;
     }
 
-    const webpPath = asset.resizePath.replace('jpeg', 'webp').replace('jpg', 'webp');
+    switch (asset.type) {
```

Comment:
Could we create another function like 'generateThumbnail(asset, kind)` which deduplicates the switch on asset type?

Reply:
Done